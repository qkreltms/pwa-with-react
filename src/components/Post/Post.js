import React from 'react'
import styles from './Post.css'
import { Link } from 'react-router-dom'

const Post = (props) => {
  return (
    <div>
      <h2 className={styles.title}>
        <Link className={styles.link} to={`/post/${props.slug}`}>{props.title}</Link></h2>
      <p className={styles.content}>{props.excerpt}</p>
    </div>
  )
}

export default Post
