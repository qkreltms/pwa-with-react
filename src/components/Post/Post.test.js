/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Post from './Post'
import toJson from 'enzyme-to-json'

test('Post component should render as expected', () => {
  const component = shallow(<Post />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
