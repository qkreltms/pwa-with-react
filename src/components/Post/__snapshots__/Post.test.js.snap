// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render as expected 1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      className="link"
      replace={false}
      to="/post/undefined"
    />
  </h2>
  <p
    className="content"
  />
</div>
`;
